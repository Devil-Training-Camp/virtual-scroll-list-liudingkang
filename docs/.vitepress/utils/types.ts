import { DefaultTheme, LocaleSpecificConfig } from 'vitepress';
export type Locales = LocaleSpecificConfig<DefaultTheme.Config> & { label: string; link?: string };
