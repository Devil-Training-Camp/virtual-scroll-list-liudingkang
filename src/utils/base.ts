export interface AnyObject {
  [key: number | string]: unknown;
}
