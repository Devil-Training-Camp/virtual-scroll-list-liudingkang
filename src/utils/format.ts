export function camelize(str: string): string {
  return str.toLocaleLowerCase();
}
