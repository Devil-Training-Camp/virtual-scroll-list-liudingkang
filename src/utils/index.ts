export * from './with-intall';
export * from './format';
export * from './optimize';
