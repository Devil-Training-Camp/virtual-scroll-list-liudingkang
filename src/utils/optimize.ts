type Timer = null | number;
export type RestFunc = <T>(...args: T[]) => void;
export function throttle<T extends RestFunc>(func: T, time = 0, immediate = true) {
  let timer: Timer = null;
  return function (this: unknown, ...args: Parameters<T>) {
    if (immediate) {
      func.call(this, ...args);
      immediate = false;
    }
    if (timer == null) {
      timer = setTimeout(() => {
        func.call(this, ...args);
        timer = null;
      }, time);
    }
  };
}
export function RAFThrottle<T extends RestFunc>(func: T) {
  let flag = false;
  return function (this: unknown, ...args: Parameters<T>) {
    if (!flag) {
      window.requestAnimationFrame(() => {
        func.call(this, ...args);
        flag = false;
      });
      flag = true;
    }
  };
}
export const BSStartIndex = (arr: number[], num: number) => {
  let left = 0,
    right = arr.length - 1;

  while (left != right) {
    const mid = left + ((right - left) >> 1);
    if (arr[mid] > num) {
      right = mid;
    } else {
      left = mid + 1;
    }
  }
  return Math.max(0, left - 1);
};
