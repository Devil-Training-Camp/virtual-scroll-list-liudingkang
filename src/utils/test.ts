import { nextTick } from 'vue';

export const delay = (time?: number) => {
  return new Promise(resolve => {
    setTimeout(resolve, time);
  });
};
export const trigger = async (target: HTMLElement, type: string) => {
  const scrollEvent = new Event(type, {
    bubbles: true,
    cancelable: true,
  });
  target.dispatchEvent(scrollEvent);
  await nextTick();
};
export const triggerScrollTo = async (target: HTMLElement, x: number = 0, y: number = 0) => {
  target.scrollTo({
    left: x,
    top: y,
  });
  await trigger(target, 'scroll');
};
