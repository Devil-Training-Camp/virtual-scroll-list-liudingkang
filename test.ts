interface VInfo {
  major: string;
  minor: string;
  patch: string;
  prerelease: string;
  preItems: string[];
  compileInfo: string;
}

function versionCompare(v1: string, v2: string): number {
  const p1 = parseV(v1);
  const p2 = parseV(v2);
  switch (finalCompare(p1, p2)) {
    case 1:
      return 1;
    case -1:
      return -1;
  }
  preCompare(p1, p2);
}
function finalCompare(p1: VInfo, p2: VInfo): number {
  const { major: major1, minor: minor1, patch: patch1 } = p1;
  const { major: major2, minor: minor2, patch: patch2 } = p2;
  switch (strNumCompare(major1, major2)) {
    case 1:
      return 1;
    case -1:
      return -1;
  }
  switch (strNumCompare(minor1, minor2)) {
    case 1:
      return 1;
    case -1:
      return -1;
  }
  switch (strNumCompare(patch1, patch2)) {
    case 1:
      return 1;
    case -1:
      return -1;
  }
  return 0;
}
function preCompare(p1: VInfo, p2: VInfo): number {
  const { preItems: preItems1 } = p1;
  const { preItems: preItems2 } = p2;
  const maxLen = Math.max(preItems1.length, preItems2.length);
  for (let i = 0; i < maxLen; i++) {
    const item1 = preItems1[i];
    const item2 = preItems2[i];
    // 有一个不存在
    if (preItems1[i].length === 0 || preItems2[i].length === 0) {
      return preItems1[i].length > 0 ? 1 : -1;
    }
    // 两个都存在
    if (isStrNum(item1) && isStrNum(item2)) {
      // 都由数字组成，比较数字大小
      switch (strNumCompare(item1, item2)) {
        case 1:
          return 1;
        case -1:
          return -1;
      }
    } else {
      // 有一个不为数字，按 ASCII 顺序比较
    }
  }
  return 0;
}
function strNumCompare(v1: string, v2: string): number {
  if (v1.length !== v2.length) {
    return v1.length > v2.length ? 1 : -1;
  }
  for (let i = 0, len = v1.length; i < len; i++) {
    if (v1[i] !== v2[i]) {
      return v1[i] > v2[i] ? 1 : -1;
    }
  }
  return 0;
}
function isStrNum(s: string): boolean {
  return s.length > 0 && !Number.isNaN(Number(s));
}

function parseV(v: string): VInfo {
  const reg = /(\d+?)\.(\d+?)\.(\d+?)-?([\w.]*)\+?([\w.]*)/;
  const matchInfo = v.match(reg);
  return {
    major: matchInfo[1],
    minor: matchInfo[2],
    patch: matchInfo[3],
    prerelease: matchInfo[4],
    preItems: matchInfo[4].split('.'),
    compileInfo: matchInfo[5],
  };
}

console.log(versionCompare('1.2.3', '9.8.7'));
