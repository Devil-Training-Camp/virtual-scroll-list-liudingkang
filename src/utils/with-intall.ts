import type { App, Component } from 'vue';
import { camelize } from './format';

export type WithInstall<T> = T & {
  install(app: App): void;
};
// 组件 install 方法
export function withInstall<T extends Component>(options: T): WithInstall<T> {
  (options as WithInstall<T>).install = (app: App) => {
    const { name } = options;
    if (!name) return;
    app.component(name, options);
    app.component(camelize(name), options);
  };
  return options as WithInstall<T>;
}
