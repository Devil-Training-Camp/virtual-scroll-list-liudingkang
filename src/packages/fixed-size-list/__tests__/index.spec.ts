import { describe, expect, test, vi } from 'vitest';
import { mount } from '@vue/test-utils';

import { FixedSizeList } from '../';
import { delay, triggerScrollTo } from '../../../utils/test';

describe('test fixed-size-list component props', () => {
  const data = new Array(10000).fill(0);
  test('test dynamic-list data 0', async () => {
    const wrapper = mount({
      props: {
        itemSize: 50,
        data: [],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('test fixed-size-list itemSize', () => {
    const wrapper = mount(FixedSizeList, {
      props: {
        itemSize: 50,
        data,
      },
    });
    expect((wrapper.find('.virtual-list-item').element as HTMLElement).style.height).toBe('50px');
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
  test('test fixed-size-list itemClass', () => {
    const wrapper = mount(FixedSizeList, {
      props: {
        data,
        itemClass: 'custom-item',
      },
    });
    expect(wrapper.find('.virtual-list-item').classes()).toContain('custom-item');
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('test fixed-size-list width and height', () => {
    const wrapper = mount(FixedSizeList, {
      props: {
        data,
        width: 300,
        height: 600,
      },
    });
    const elm = wrapper.find('.virtual-list-container').element as HTMLElement;
    expect(elm.style.width).toBe('300px');
    expect(elm.style.height).toBe('600px');
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('test fixed-size-list cache', () => {
    const wrapper = mount(FixedSizeList, {
      props: {
        data,
        height: 600,
        itemSize: 155,
        cache: 2,
      },
    });
    expect(wrapper.findAll('.virtual-list-item')).toHaveLength(6);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
describe('test fixed-size-list scroll', () => {
  test('test fixed-size-list scroll cache', async () => {
    const data = new Array(20).fill(0);

    const wrapper = mount(FixedSizeList, {
      props: {
        data,
        height: 600,
        itemSize: 150,
        cache: 4,
      },
    });
    const elm = wrapper.find('.virtual-list-container').element as HTMLElement;
    expect(wrapper.findAll('.virtual-list-item')).toHaveLength(8);
    expect(wrapper.html()).toMatchSnapshot();
    await triggerScrollTo(elm, 0, 650);
    await delay();
    expect(wrapper.findAll('.virtual-list-item')).toHaveLength(12);
    expect(wrapper.html()).toMatchSnapshot();
    await triggerScrollTo(elm, 0, 850);
    await delay();
    expect(wrapper.findAll('.virtual-list-item')).toHaveLength(12);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('test fixed-size-list scroll distance', async () => {
    const data = new Array(5).fill(0);
    const onLoad = vi.fn();

    const wrapper = mount(FixedSizeList, {
      props: {
        data,
        height: 600,
        itemSize: 150,
        distance: 100,
        onLoad,
      },
    });
    const elm = wrapper.find('.virtual-list-container').element as HTMLElement;
    await triggerScrollTo(elm, 0, 50);
    await delay();
    expect(onLoad).toBeCalledTimes(1);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
describe('test fixed-size-list event', () => {
  test('test fixed-size-list onload', async () => {
    const data = new Array(5).fill(0);
    const onLoad = vi.fn();

    const wrapper = mount(FixedSizeList, {
      props: {
        data,
        height: 600,
        itemSize: 150,
        onLoad,
      },
    });
    const elm = wrapper.find('.virtual-list-container').element as HTMLElement;
    await triggerScrollTo(elm, 0, 150);
    await delay();
    expect(onLoad).toBeCalledTimes(1);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
