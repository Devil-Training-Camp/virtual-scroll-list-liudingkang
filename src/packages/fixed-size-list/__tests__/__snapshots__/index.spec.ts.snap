// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test fixed-size-list component props > test dynamic-list data 0 1`] = `""`;

exports[`test fixed-size-list component props > test fixed-size-list cache 1`] = `
"<div data-v-a7e82dad="" class="ldk-virtual-list-container ldk-fixed-size-list" style="width: 100%; height: 600px;">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 1550000px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="0" style="top: 0px; height: 155px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 155px; height: 155px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 310px; height: 155px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 465px; height: 155px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 620px; height: 155px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="5" style="top: 775px; height: 155px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list component props > test fixed-size-list itemClass 1`] = `
"<div data-v-a7e82dad="" class="ldk-virtual-list-container ldk-fixed-size-list" style="width: 100%; height: 300px;">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 700000px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item custom-item" data-index="0" style="top: 0px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item custom-item" data-index="1" style="top: 70px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item custom-item" data-index="2" style="top: 140px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item custom-item" data-index="3" style="top: 210px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item custom-item" data-index="4" style="top: 280px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item custom-item" data-index="5" style="top: 350px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item custom-item" data-index="6" style="top: 420px; height: 70px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list component props > test fixed-size-list itemSize 1`] = `
"<div data-v-a7e82dad="" class="ldk-virtual-list-container ldk-fixed-size-list" style="width: 100%; height: 300px;">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 500000px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="0" style="top: 0px; height: 50px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 50px; height: 50px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 100px; height: 50px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 150px; height: 50px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 200px; height: 50px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="5" style="top: 250px; height: 50px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="6" style="top: 300px; height: 50px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="7" style="top: 350px; height: 50px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list component props > test fixed-size-list width and height 1`] = `
"<div data-v-a7e82dad="" class="ldk-virtual-list-container ldk-fixed-size-list" style="width: 300px; height: 600px;">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 700000px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="0" style="top: 0px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 70px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 140px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 210px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 280px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="5" style="top: 350px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="6" style="top: 420px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="7" style="top: 490px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="8" style="top: 560px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="9" style="top: 630px; height: 70px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="10" style="top: 700px; height: 70px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list event > test fixed-size-list onload 1`] = `
"<div data-v-a7e82dad="" style="width: 100%; height: 600px;" class="ldk-virtual-list-container ldk-fixed-size-list ldk-virtual-list-scrolling">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 750px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="0" style="top: 0px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 150px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 300px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 450px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 600px; height: 150px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list scroll > test fixed-size-list scroll cache 1`] = `
"<div data-v-a7e82dad="" class="ldk-virtual-list-container ldk-fixed-size-list" style="width: 100%; height: 600px;">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 3000px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="0" style="top: 0px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 150px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 300px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 450px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 600px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="5" style="top: 750px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="6" style="top: 900px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="7" style="top: 1050px; height: 150px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list scroll > test fixed-size-list scroll cache 2`] = `
"<div data-v-a7e82dad="" style="width: 100%; height: 600px;" class="ldk-virtual-list-container ldk-fixed-size-list ldk-virtual-list-scrolling">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 3000px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="0" style="top: 0px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 150px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 300px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 450px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 600px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="5" style="top: 750px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="6" style="top: 900px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="7" style="top: 1050px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="8" style="top: 1200px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="9" style="top: 1350px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="10" style="top: 1500px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="11" style="top: 1650px; height: 150px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list scroll > test fixed-size-list scroll cache 3`] = `
"<div data-v-a7e82dad="" style="width: 100%; height: 600px;" class="ldk-virtual-list-container ldk-fixed-size-list ldk-virtual-list-scrolling">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 3000px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 150px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 300px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 450px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 600px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="5" style="top: 750px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="6" style="top: 900px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="7" style="top: 1050px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="8" style="top: 1200px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="9" style="top: 1350px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="10" style="top: 1500px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="11" style="top: 1650px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="12" style="top: 1800px; height: 150px;"></div>
  </div>
</div>"
`;

exports[`test fixed-size-list scroll > test fixed-size-list scroll distance 1`] = `
"<div data-v-a7e82dad="" style="width: 100%; height: 600px;" class="ldk-virtual-list-container ldk-fixed-size-list ldk-virtual-list-scrolling">
  <div data-v-a7e82dad="" class="ldk-virtual-list" style="width: 100%; height: 750px;">
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="0" style="top: 0px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="1" style="top: 150px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="2" style="top: 300px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="3" style="top: 450px; height: 150px;"></div>
    <div data-v-e96bb3d2="" data-v-a7e82dad="" class="ldk-virtual-list-item" data-index="4" style="top: 600px; height: 150px;"></div>
  </div>
</div>"
`;
